roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-client-element.conf.j2

```diff
@@ -10,6 +10,12 @@
 	{% if matrix_nginx_proxy_floc_optout_enabled %}
 		add_header Permissions-Policy interest-cohort=() always;
 	{% endif %}
+	
+	{% if matrix_nginx_proxy_ocsp_stapling_enabled and matrix_ssl_retrieval_method in ["lets-encrypt", "manually-managed"] %}
+		ssl_stapling on;
+		ssl_stapling_verify on;
+		ssl_trusted_certificate {{ matrix_ssl_config_dir_path }}/live/{{ matrix_nginx_proxy_base_domain_hostname }}/chain.pem;
+	{% endif %}
```

Comment:
This likely applies to all other vhosts too.

I see you've added this new configuration inside `{% macro render_vhost_directives() %}`.

This **usually** gets rendered into an SSL-enabled vhost, but it's not always the case.

If `matrix_nginx_proxy_https_enabled` is `false`, then `render_vhost_directives()` gets rendered into a plain-HTTP `server` block. See here: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/e4dd933cf0f48a8722a43239b07a9335c85c4f14/roles/matrix-nginx-proxy/templates/nginx/conf.d/matrix-client-element.conf.j2#L42-L60

-------------

Not sure if having `ssl_*` directives in non-SSL-enabled `server` block causes critical nginx errors (maybe it does). Even if it doesn't, it's still not very clean to do it there.

I suggest all these get moved down to the SSL-enabled `server` block, next to all the other `ssl_*` directives.

Reply:
yeah right thanks 👍